bankingapp/handler/account/bankaccounthandler.go

```diff
@@ -203,20 +203,20 @@ func Withdraw(ctx *gin.Context) {
 	defer tx.Rollback()
 
 	// Confirm that account have enough funds.
-	var enough bool
-	var openingBalance float64
+	var enough, overdraftallowed bool
+	var openingBalance, overdraftamount, newBalance, newOdAmount float64
 	var customerId int64
 	//changes for fixed a/c withdrawl
 	var accType string
 	var isAccActive int
 	var lockPeriodFD string
-	var fdPenalty float32
+	var fdPenalty float64
 	var lockedUntil string
 	if err = tx.QueryRowContext(ctx, `SELECT (balance >= ?), balance, customer_id, account_type, is_active, 
-	IFNULL(lock_period_fd, ''), penalty_fd, IFNULL(locked_until, '')
+	IFNULL(lock_period_fd, ''), penalty_fd, IFNULL(locked_until, ''), (odallowed=true), odamount 
```

Comment:
(odallowed=true) this will always return true, did you mean (odallowed==true)?

Reply:
changed